packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -317,13 +324,21 @@ export class HbarLimitService implements IHbarLimitService {
   }
 
   /**
-   * Gets a new reset timestamp for the rate limiter.
+   * Gets the new reset timestamp for the rate limiter.
+   *
+   * If the new reset timestamp falls on a different day, the time is set to midnight.
+   *
    * @returns {Date} - The new reset timestamp.
    * @private
    */
   private getResetTimestamp(): Date {
-    const tomorrow = new Date(Date.now() + HbarLimitService.ONE_DAY_IN_MILLIS);
-    return new Date(tomorrow.setHours(0, 0, 0, 0));
+    const now = new Date();
+    const resetDate = new Date(now.getTime() + this.limitDuration);
+    if (this.limitDuration >= this.oneDayInMillis) {
```

Comment:
Q: Is this correct?
Does this mean our intention is that the hbar limit at max supports logic up to on day?
Just need to confirm and be explicit in the docs if so how one day and the other durations interact with each other

Reply:
It supports more than one day as well, the idea is described in the jsdocs: `If the new reset timestamp falls on a different day, the time is set to midnight.`, so what that means is:
- we start-up the relay at let's say 3:00PM
- if we set the duration to 2 days, the reset date will be after 2 days at midnight
- if we set the duration to 1 day, the reset date will be after 1 day at midnight
- if we set the duration to 4 hours, the reset date will be after 4 hours

Reply:
Changed the way it is calculated, and also added docs:
```typescript
  /**
   * Calculates the next reset timestamp for the rate limiter.
   *
   * This method determines the next reset timestamp based on the current reset timestamp
   * and the limit duration. If the current reset timestamp is not defined, it initializes
   * the reset timestamp to midnight of the current day. It then iteratively adds the limit
   * duration to the reset timestamp until it is in the future.
   *
   * @returns {Date} - The next reset timestamp.
   */
  private getResetTimestamp(): Date {
    const todayAtMidnight = new Date().setHours(0, 0, 0, 0);

    let newResetDate = this.reset !== undefined ? new Date(this.reset.getTime()) : new Date(todayAtMidnight);
    while (newResetDate.getTime() < Date.now()) {
      newResetDate = new Date(newResetDate.getTime() + this.limitDuration);
    }

    return newResetDate;
  }
```